feat: release 2.10.3

bzp2010 opened the pull request:
### What this PR does / why we need it:
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

Update `rockspec` and `CHANGELOG` for version 2.10.3. And updates related to version code and documentation will be provided in the PR of `release/2.10` to prevent affecting the master branch.

### Pre-submission checklist:

<!--
Please follow the requirements:
1. Use Draft if the PR is not ready to be reviewed
2. Test is required for the feat/fix PR, unless you have a good reason
3. Doc is required for the feat PR
4. Use a new commit to resolve review instead of `push -f`
5. If you need to resolve merge conflicts after the PR is reviewed, please merge master but do not rebase
6. Use "request review" to notify the reviewer once you have resolved the review
-->

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

juzhiyuan commented:
Hi @bzp2010, do we need this? https://github.com/apache/apisix/pull/5958

bzp2010 commented:
> Hi @bzp2010, do we need this? #5958

@juzhiyuan They will be pushed to a different branch. The use of this PR for pushing to master branch can be found in the following PR. In this release, some of the changes involved in the following PR [1] will be split into two parts to prevent affecting the master branch code.

[1] https://github.com/apache/apisix/pull/5508

spacewander reviewed (changes requested):
Merge this changelog will confuse the user about the timeline. Those fixes are in 2.10.3 but not in 2.11.0. Also, when we release 2.12.0, we have to mention some of those fixes again.

And this PR doesn't update the Chinese version of CHANGELOG.

spacewander commented on CHANGELOG.md:
```diff
@@ -79,6 +80,26 @@ title: Changelog
+- fix(patch): add global `math.randomseed` patch support [#5682](https://github.com/apache/apisix/pull/5682)
+- fix(log-rotate): after enabling compression collect log exceptions [#5715](https://github.com/apache/apisix/pull/5715)
+- feat(ext-plugin): avoid sending conf request more times [#5183](https://github.com/apache/apisix/pull/5183)
+- feat: use lock to ensure fetching token from shdict always [#5263](https://github.com/apache/apisix/pull/5263)
```
Look like some features are incorrectly backport as bugfix?

bzp2010 commented:
> Merge this changelog will confuse the user about the timeline. Those fixes are in 2.10.3 but not in 2.11.0. Also, when we release 2.12.0, we have to mention some of those fixes again.
> 
> And this PR doesn't update the Chinese version of CHANGELOG.

Hi, @spacewander.

I added a partial description of what PRs were merged into this "LTS maintenance release" for version 2.10.3 as shown below, does this ease the confusion? Or do I need to remove CHANGELOG and make it provide only in the 2.10 branch. (But a CHANGELOG in the master branch is also need? How should I do it?)

Also, I have added the Chinese version of CHANGELOG.

spacewander commented:
@bzp2010 
Your solution doesn't solve the problem I mention...

spacewander commented on rockspec/apisix-2.10.3-0.rockspec:
```diff
@@ -0,0 +1,95 @@
+--
```
Maybe we don't need to archive this rockspec? We don't merge the code into the default branch.

bzp2010 replied:
@spacewander Here it is marked for installation using the `2.10.3` git tag, as far as I know this could potentially be used when packaging various container images?

[https://github.com/apache/apisix-docker/blob/5d3c4c63d21f0c40655d005fdb9cb3746f2e4af3/alpine/Dockerfile#L22](https://github.com/apache/apisix-docker/blob/5d3c4c63d21f0c40655d005fdb9cb3746f2e4af3/alpine/Dockerfile#L22)

bzp2010 replied:
Hi, @spacewander. Do I need to make changes for this?

spacewander replied:
I see.

bzp2010 commented on docs/zh/latest/CHANGELOG.md:
```diff
@@ -79,34 +80,23 @@ title: CHANGELOG
+## 2.10.3
 
-### Bugfix
+**这是一个LTS维护版本，您可以在 `release/2.10` 分支中看到CHANGELOG。**
```
updated
